rotkehlchen/externalapis/beaconchain/service.py

```diff
@@ -96,10 +97,15 @@ def _query(
         retries_num = times
         timeout = (CachedSettings().get_timeout_tuple()[0], BEACONCHAIN_READ_TIMEOUT)
         backoff_in_seconds = 10
-        log.debug(f'Querying beaconcha.in API for {query_str}')
+        log.debug(f'Querying beaconcha.in API for {query_str} with {data=}')
         while True:
             try:
-                response = self.session.get(query_str, timeout=timeout)
+                response = self.session.request(
+                    method='POST' if endpoint is None and module == 'validator' else 'GET',
```

Comment:
same comment as before here for the if condition and instead moving this as an argument